src/node_sqlite.cc

```diff
@@ -653,6 +837,79 @@ BaseObjectPtr<StatementSync> StatementSync::Create(Environment* env,
   return MakeBaseObject<StatementSync>(env, obj, db, stmt);
 }
 
+Session::Session(Environment* env,
+                 Local<Object> object,
+                 BaseObjectWeakPtr<DatabaseSync> database,
+                 sqlite3_session* session)
+    : BaseObject(env, object),
+      session_(session),
+      database_(std::move(database)) {
+  MakeWeak();
+}
+
+Session::~Session() {
+  if (!database_ || !database_->connection_) return;
+  sqlite3session_delete(session_);
+  database_->sessions_.erase(session_);
+}
+
+BaseObjectPtr<Session> Session::Create(Environment* env,
+                                       BaseObjectWeakPtr<DatabaseSync> database,
+                                       sqlite3_session* session) {
+  Local<Object> obj;
+  if (!GetConstructorTemplate(env)
+           ->InstanceTemplate()
+           ->NewInstance(env->context())
+           .ToLocal(&obj)) {
+    return BaseObjectPtr<Session>();
+  }
+
+  return MakeBaseObject<Session>(env, obj, std::move(database), session);
+}
+
+Local<FunctionTemplate> Session::GetConstructorTemplate(Environment* env) {
+  Local<FunctionTemplate> tmpl = env->sqlite_session_constructor_template();
+  if (tmpl.IsEmpty()) {
+    Isolate* isolate = env->isolate();
+    tmpl = NewFunctionTemplate(isolate, IllegalConstructor);
+    tmpl->SetClassName(FIXED_ONE_BYTE_STRING(env->isolate(), "Session"));
+    tmpl->InstanceTemplate()->SetInternalFieldCount(
+        Session::kInternalFieldCount);
+    SetProtoMethod(isolate,
+                   tmpl,
+                   "changeset",
+                   Session::Changeset<sqlite3session_changeset>);
+    SetProtoMethod(
+        isolate, tmpl, "patchset", Session::Changeset<sqlite3session_patchset>);
+    env->set_sqlite_session_constructor_template(tmpl);
+  }
+  return tmpl;
+}
+
+void Session::MemoryInfo(MemoryTracker* tracker) const {}
```

Comment:
I am not sure how to implement these.

Reply:
@jasnell Do you know if anything needs to happen here?

Reply:
For now I'd say it is ok to leave this as is.